packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -748,6 +778,12 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
     }
   }
 
+  private handleClear = () => {
+    const { multiple } = this.props
+
+    this.setState({ value: multiple ? [] : '' })
```

Comment:
this won't be enough for `search` dropdowns, you need to reset the `searchQuery` as well
```suggestion
    this.setState({ value: multiple ? [] : '', searchQuery: search ? '' : undefined })
```
I'd suggest also extracting the logic from `getInitialAutoControlledState` into a common function that we'd call in both places

Reply:
Reused values from ``getInitialAutoControlledState()`, nice suggestion 👍